File: DSA/fundamentals/arrays_and_lists/part0/python/judge.py
from solution import Solution

class Judge:
    def run_tests(self):
        test_cases_a = [
            [1, 2, 3, 4, 5],
            [1, 2, 3, 4, 5],
        ]
        test_cases_k = [
            1,
            0,
        ]

        passes = 0;
        for i in range(len(test_cases_a)):
            if self.test(test_cases_a[i].copy(), test_cases_k[i], self.test_case_solution_output(test_cases_a[i].copy(), test_cases_k[i])):
                passes += 1
        print(f"Passed {passes} of {len(test_cases_a)} tests.")

    def test(self, a, k, expected):
        solution = Solution()
        solution.increment_array(a, k)
        passed = a == expected

        print(f"a: {a}")
        print(f"k: {k}")
        print(f"Expected Output: {expected}")
        print(f"Your Output: {a}")
        print("Test " + ("PASSED" if passed else "FAILED"))
        print("-----------")
        return passed

    def test_case_solution_output(self, a, k):
        for i in range(len(a)):
            a[i] += k
        return a

File: DSA/fundamentals/arrays_and_lists/part0/javascript/judge.js
const Solution = require('./solution');

class Judge {
    runTests() {
        const testCasesA = [
            [1, 2, 3, 4, 5],
            [1, 2, 3, 4, 5],
        ];
        const testCasesK = [
            1,
            0,
        ];

        let passes = 0;
        for (let i = 0; i < testCasesA.length; i++) {
            if (this.test(Array.from(testCasesA[i]), testCasesK[i], this.testCaseSolutionOutput(Array.from(testCasesA[i]), testCasesK[i]))) {
                passes++;
            }
        }
        console.log(`Passed ${passes} of ${testCasesA.length} tests.`);
    }

    test(a, k, expected) {
        const solution = new Solution();
        solution.incrementArray(a, k);
        const passed = JSON.stringify(a) === JSON.stringify(expected);

        console.log(`a: ${JSON.stringify(a)}`);
        console.log(`k: ${k}`);
        console.log(`Expected Output: ${JSON.stringify(expected)}`);
        console.log(`Your Output: ${JSON.stringify(a)}`);
        console.log(passed ? "Test PASSED" : "Test FAILED");
        console.log("-----------");
        return passed;
    }

    testCaseSolutionOutput(a, k) {
        for (let i = 0; i < a.length; i++) {
            a[i] += k;
        }
        return a;
    }
}

module.exports = Judge;